### PR TITLE
#164829380 Adds feature to highlight and comment on any text in an article

### PR DESCRIPTION
**Pull request by Raywire**
**What does this PR do?**
Adds a feature to highlight a text on an article and comment on the highlight.

**Description of the tasks to be completed**

1. Add tests.

2. Add functionality to Highlight a text on an article.

3. Add functionality to comment on an article highlight.

4. Add functionality to delete a highlight.

5. Add functionality to get all highlights of a given article.

**How should this be manually tested?**

- Open your terminal window and clone the repo,  and navigate to the root directory.
- Fetch all the branches and checkout into this feature's branch using the commands below
```bash
git fetch --all
git checkout ft-highlight-and-comment-164829380
```
- Create a virtual environment and then activate the environment as shown below
```python
virtualenv venv
source venv/bin/activate
```
- Install all the required dependencies using pip by running 
```python
pip install -r requirements.txt
```
- Run the Django server to serve up the app by typing 
```python
python manage.py runserver
```
**Highlighting a text on an article**

Navigate to the URL below on postman and send a POST request as shown on the screenshot below.
`http://localhost/api/articles/<slug>/highlight/`

The payload should be in the format shown below where `start` stands for the first position where the highlight starts and  `end ` stands for the end position of the highlight. All these fields should be positive integers.
```
{
	"highlight":{
		"start": positive integer,
		"end": positive integer,
		
	}
}
```

Screenshot 2019-04-23 at 11 25 29

**Retrieve all highlights for an article**

Send a GET request containing the slug for the article you want to retrieve its highlights.
```json
`http://localhost/api/articles/<slug>/highlight/`
```
Screenshot 2019-04-23 at 11 37 20

**Commenting on an article**

Using Postman, send a POST request containing the slug of the article and the payload should contain a comment field as shown below.
```json
http://localhost/api/articles/<slug>/image/<id>/
```
Screenshot 2019-04-23 at 11 40 51

_**Notes**_
Highlighting a text twice will automatically delete the highlight.

**Relevant PT stories**
https://www.pivotaltracker.com/story/show/164829380

### CODE REVIEW

**Review by MainaKamau92 — DISMISSED** [review]
Good work on being able to catch a highlight endpoint that exceeds the length of the article

File: authors/apps/articles/views.py [hunk 1]
```diff
@@ -44,6 +44,48 @@ def find_favorite(slug):
...
 
+def get_highlights(slug):
+    """Method to get all highlights of an article by slug"""
+    return Highlight.objects.select_related(
+            'article').filter(article__slug=slug)
+
+
+def format_highlight(highlights, saved_article):
+    """Method to update start and end index of highlight if article
+    body is updated or deleted the highlight if idtdoes not exist anymore"""
```
Comment (kirega) [inline authors/apps/articles/views.py]: You have a minor typo here.

Comment (Raywire) [reply authors/apps/articles/views.py]: @kirega Thanks for the feedback the typo has been corrected.

**Review by MainaKamau92 — APPROVED** [review]

**Comment by kirega** [comment]
Good job on this task. The exceptions are handled gracefully and most importantly everything works.
The edge cases also pass.

**Review by kirega — APPROVED** [review]

**Pull request merged**